contentcuration/contentcuration/static/js/edit_channel/uploader/components/AssessmentItemToolbar/__snapshots__/AssessmentItemToolbar.spec.js.snap

```diff
@@ -38,89 +15,28 @@ exports[`AssessmentItemToolbar doesn't render move up action in menu if it shoul
   <div class="flex toolbar-item">
     <div class="v-menu"><button type="button" class="v-btn v-btn--icon theme--light">
         <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light grey--text text--darken-1">more_vert</i></div>
-      </button>
-      <div class="v-menu__content theme--light " style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;">
-        <div role="list" class="v-list theme--light">
-          <div role="listitem"><a data-test="toolbarMenuItem-MOVE_ITEM_DOWN" class="v-list__tile v-list__tile--link theme--light">
-              <div class="v-list__tile__title">Move down</div>
-            </a></div>
-        </div>
-        <div role="list" class="v-list theme--light">
-          <div role="listitem"><a data-test="toolbarMenuItem-ADD_ITEM_ABOVE" class="v-list__tile v-list__tile--link theme--light">
-              <div class="v-list__tile__title">Add item above</div>
-            </a></div>
-        </div>
-        <div role="list" class="v-list theme--light">
-          <div role="listitem"><a data-test="toolbarMenuItem-ADD_ITEM_BELOW" class="v-list__tile v-list__tile--link theme--light">
-              <div class="v-list__tile__title">Add item below</div>
-            </a></div>
-        </div>
-        <div role="list" class="v-list theme--light">
-          <div role="listitem"><a data-test="toolbarMenuItem-DELETE_ITEM" class="v-list__tile v-list__tile--link theme--light">
-              <div class="v-list__tile__title">Delete</div>
-            </a></div>
-        </div>
-      </div>
-    </div>
+      </button></div>
```

Comment:
For what it's worth, I have never seen a "snaphot test" catch a real bug in Kolibri, but they will very regularly break due to intentional changes. CC @jonboiser @rtibbles I'm curious if you guys have had the same experiences?

A good unit test will be say something about what "should" be true, based on the particular intent of the software developer. For example, they might say "if file uploads exist, and two are currently in progress, we should see exactly two loading indicators."

Snapshot tests on the other hand dump the pure DOM output, with no semantically meaningful test about how the UI should look. This makes them:

* very brittle because all legitimate changes will break the snapshot
* useless as tests because they don't say anything about the way the UI should actually look or behave

I would recommend considering _not_ using snapshot tests, or at the very least they should not be a replacement for real, meaningful tests.

Reply:
That makes sense- since this is outside of the scope of this PR, I'll flag this comment for @MisRob to revisit during the next iteration on the exercise editor

Reply:
I agree with what unit test is supposed to do. Though I am not strictly against using snapshots when it comes to simpler, rather visual checks that all information more or less shows up for small to medium size pure UI components until we have a better tool for that. It's difficult to maintain tons of unit tests to check that buttons have proper text, all data are rendered in a table and similar. I actually don't classify such checks as unit tests. But it's definitely true that even for smaller components, especially those that have the potential to be updated often, maintenance is difficult.

No problem to remove snapshot tests mentioned in this PR when I am back working on exercises - haven't checked them in detail but diff seems to be annoying indeed. Just thinking that there's no need to close the door to snapshots completely - but totally agree that it's something to be used with caution.

I am thinking that I could incorporate this issue into Storybook introduction as for me it's a more comfortable tool for visual checks anyway so let's see if it could fit our use cases :)

Reply:
I've just added the task to my list of follow-ups - [cleanup section](https://www.notion.so/learningequality/Exercise-creation-1a085a11eb494ccab3962a22c20dd202#d763bed3a3294672861eb8d42e1c9228)